qmd/wide-names.qmd

```diff
@@ -0,0 +1,132 @@
+
+## PHES-ODM approach to wide- and long-formats
+
+The PHES-ODM is designed as a series of long-format tables, but aims to provide robust infrastructure to support wide-formats. This support includes managing mapping between long- and wide-formats. To streamline mapping between formats, we have created a standardized approach to generating wide-format column names (or wide-names).
+
+![**Figure 1 - Wide- and Long-names**](assets/Long-to-wide-name%20example.png){fig-align="center"}
+
+For long-formatted data in the ODM, names are all made using camel case (ex: `exampleName`). Each long format name is a 'part', which are the building blocks of the data model. All wide-format names are snake case (ex: `example_name`). Each section of the snake case wide name is an individual part that is combined to form the fully-specified wide-name. Using this approach, wide-names can be parsed automatically to move and transform data between formats without introducing errors.
+
+# Wide-names
+
+The exact formula used to generate a wide-name in the ODM depends on the kind of data point you are recording. The three kinds of data and metadata for these column names are either attributes, measures, or protocol steps. An attribute column is metadata. These columns provide context about a measure, protocol, or sample. Measure columns report a specific measure, either from a site, sample, or population. Lastly, protocol columns report a specific protocol step or detail for a measure.
+
+## Attributes
+
+The wide-name formula for attributes is `tableShortName_attribute`.
+
+For example, to record the sample material in long-format you would navigate to the `samples` table and populate the `saMaterial` column. In wide-format, all data are recorded in a single table. The table and attribute are combined for the wide-name, creating the name `sm_saMaterial` for this attribute. Here `sm` is the short name for the samples table, and `saMaterial` is the column name for the sample matrix or material.
+
+## Measures
+
+The wide-name formula for a measure or measure-value is `compartment_specimen_fraction_measure_unit_aggregation_index_attribute`, where the `attribute` at the end is always either `value`, `purpose`, or `qualityFlag`.
+
+The ODM long-format specifies a `measure` column in the `measures` table, as well as a column for `units` and one for `aggregations`. The wide-format collapses all these columns into a single `value` column.
+
+For example, the wide-name for mean chemical oxygen demand from a wastewater sampling site, recorded in milligrams per litre is `wat_si_NR_cod_mgL_m_NR_value`. Here the measure is taken from wastewater/water (`wat`), it's a measure of the site (`si`), the fraction analyzed is not reported (`NR`), it is a measure of chemical oxygen demand (`cod`), the units are mg/L (`mgL`), it's a mean measure (`m`), the index is not reported (`NR`), and lastly this is the value for this measure (`value`).
```

Comment:
```suggestion
| date | covid N1 (gc/mL) | PMMoV (gc/mL) | pH (unitless)|
```

Reply:
I added some example figures that illustrate the examples more - can you take a look and see if you think this covers off some of the examples?